### PR TITLE
HO: Fix haproxy to have apiserver external address for public clusters

### PR DESCRIPTION
**Pull request by alvaroaleman**
This was accidentally removed in
https://github.com/openshift/hypershift/pull/1614/files#diff-8d5b507fe367a80159912e2310bc3e7b76a5aa9dd1f12ccee7a1bf3a9bc2ac49
and breaks public clusters.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/nodepool/haproxy.go [hunk 1]
```diff
@@ -62,6 +65,27 @@ func (r *NodePoolReconciler) reconcileHAProxyIgnitionConfig(ctx context.Context,
 	apiServerExternalPort := util.APIPortWithDefaultFromHostedCluster(hcluster, config.DefaultAPIServerPort)
 	if util.IsPrivateHC(hcluster) {
 		apiServerExternalAddress = fmt.Sprintf("api.%s.hypershift.local", hcluster.Name)
+	} else {
```
Comment (enxebre) [inline hypershift-operator/controllers/nodepool/haproxy.go:68]: can we have a unit tested getApiServerExternalAddress so we don't drop by accident again?

**Comment by enxebre** [comment]
/lgtm
so we fix broken public and follow up with test.

File: hypershift-operator/controllers/nodepool/haproxy.go (hunk 1 above)
Comment (alvaroaleman) [reply hypershift-operator/controllers/nodepool/haproxy.go:68]: https://github.com/openshift/hypershift/pull/1652

**Pull request merged**